sdk/core/azure-core-amqp/src/amqp/claim_based_security.cpp

```diff
@@ -134,9 +135,58 @@ namespace Azure { namespace Core { namespace Amqp { namespace _detail {
         default:
           throw std::runtime_error("Unknown management operation status.");
       }
+      Log::Stream(Logger::Level::Informational)
+          << "CBS PutToken result: " << cbsResult << " status code: " << result.StatusCode
+          << " Error: " << result.Error.Description << ".";
       return std::make_tuple(cbsResult, result.StatusCode, result.Error.Description);
     }
   }
+  std::ostream& operator<<(std::ostream& os, CbsOperationResult const& operationResult)
+  {
+    switch (operationResult)
+    {
+      case CbsOperationResult::Invalid:
+        os << "Invalid";
+        break;
+      case CbsOperationResult::Ok:
+        os << "Ok";
+        break;
+      case CbsOperationResult::Error:
+        os << "Error";
+        break;
+      case CbsOperationResult::Failed:
+        os << "Failed";
+        break;
+      case CbsOperationResult::InstanceClosed:
+        os << "InstanceClosed";
+        break;
+      default:
+        os << "Unknown CbsOperationResult." << static_cast<int>(operationResult);
```

Comment:
nit: use `std::to_string(static_cast<std::underlying_type<X>::type>(x))`: https://godbolt.org/z/dq1E89GEq

Reply:
Why use `std::to_string`" rather than the `ostream` inserter for int? I would think that the ostream inserter for int would almost always be more efficient than converting the int to a std::string then calling the ostream inserter for std::string. You can see this in the godbolt listing that you linked above. 

I can see the `std::underlying_type` construct as defensive coding against [dcl.enum#7](https://timsong-cpp.github.io/cppwp/n4659/dcl.enum#7), although none of the compilers we test against use a type other than signed int.

Reply:
The only combination that produced correct results in my example was the `std::to_string(static_cast<std::underlying_type<X>::type>(x))`:
```cpp
#include <iostream> // cout
#include <cstdint> // intX_t
#include <limits>

#include <type_traits> // underlying_type
#include <string> // to_string

enum class U8 : std::uint8_t
{
    A = 36
};

enum class U64 : std::uint64_t
{
        
    A = 18446744073709551615ULL
};

enum class S64 : std::int64_t
{
        
    A = -9223372036854775807LL - 1LL
};

int main()
{
    std::cout << "\n";
    std::cout << static_cast<int>(U8::A) << "\n"; // prints 36
    std::cout << static_cast<int>(U64::A) << "\n"; // prints -1, incorrect
    std::cout << static_cast<int>(S64::A) << "\n"; // prints 0, incorrect
    std::cout << "\n";
    std::cout << static_cast<std::int64_t>(U8::A) << "\n"; // prints 36
    std::cout << static_cast<std::int64_t>(U64::A) << "\n"; // prints -1, incorrect
    std::cout << static_cast<std::int64_t>(S64::A) << "\n"; // prints -9223372036854775808
    std::cout << "\n";
    std::cout << static_cast<std::uint64_t>(U8::A) << "\n"; // prints 36
    std::cout << static_cast<std::uint64_t>(U64::A) << "\n"; // prints 18446744073709551615
    std::cout << static_cast<std::uint64_t>(S64::A) << "\n"; // prints 9223372036854775808, incorrect
    std::cout << "\n";
    std::cout << static_cast<std::underlying_type<U8>::type>(U8::A) << "\n"; // prints $, incorrect
    std::cout << static_cast<std::underlying_type<U64>::type>(U64::A) << "\n"; // prints 18446744073709551615
    std::cout << static_cast<std::underlying_type<S64>::type>(S64::A) << "\n"; // prints -9223372036854775808
    std::cout << "\n";
    std::cout << std::to_string(static_cast<std::underlying_type<U8>::type>(U8::A)) << "\n"; // prints 36, CORRECT
    std::cout << std::to_string(static_cast<std::underlying_type<U64>::type>(U64::A)) << "\n"; // prints 18446744073709551615, CORRECT
    std::cout << std::to_string(static_cast<std::underlying_type<S64>::type>(S64::A)) << "\n"; // prints -9223372036854775808, CORRECT
}
```

Reply:
https://stackoverflow.com/questions/36843616/why-stdiostream-treat-signed-unsigned-char-as-a-text-and-not-an-integer

The standard says that std::int8_t and std::uint8_t are character types. That is explicitly why C++17 created `std::byte`.

FWIW, that means that no standards conforming compiler will ever chose a char, unsigned char, uint8_t or int8_t as the underlying representation for an enumeration because those are defined as character types and not integer types (the standard requires that the underlying type of an enumeration be an integer type).